tests/FSharp.Compiler.ComponentTests/ErrorMessages/TailCallAttribute.fs

```diff
@@ -1477,15 +1469,7 @@ namespace N
         |> withLangVersionPreview
         |> compile
         |> shouldFail
-        |> withResults [
-            { Error = Warning 3861
-              Range = { StartLine = 7
-                        StartColumn = 17
-                        EndLine = 7
-                        EndColumn = 37 }
-              Message =
-                           "The TailCall attribute should only be applied to recursive functions." }
-        ]
+        |> withSingleDiagnostic (Error 842, Line 6, Col 11, Line 6, Col 19, "This attribute is not valid for use on this language element")
```

Comment:
This would be a breaking change currently, previous message was much clearer. Is it possible we have an exception for specifically this tail call attribute on rec vals, or will it be too ad-hoc?

Reply:
> This would be a breaking change currently, previous message was much clearer. Is it possible we have an exception for specifically this tail call attribute on rec vals, or will it be too ad-hoc?

Personally I think this is right way of enforcing the correct `AttributeTarget` adding a custom exception will a third way do checking is let binding is a function. 

I guess what we can do ir to keep the previous logic and only use the error in F# 9 using the language version ?

Reply:
As per Edgar's comment, I hope this should be clear as it is now.